index.html

```diff
@@ -68,18 +68,34 @@ <h2 id="introduction">Introduction</h2>
+
+<p>The <a>bidi algorithm</a> is primarily focused on arranging adjacent characters, based on character properties. The base direction dictates (a) the visual order and direction in which runs of strongly-typed LTR and RTL character are displayed, and (b) where there are weakly-directional or neutral characters such as punctuation, the placement of those items relative to the other content.</p>
+ 
+<p class="definition"><dfn>Bidi isolation</dfn> often needs to be applied to a range of text in order to prevent the automatic rules of the Unicode Bidirectional Algorithm incorrectly ordering that content in relation to the surrounding text.</p>
+
+<p>For example, numbers following right-to-left text in memory are automatically positioned to the left of that text by the Bidi Algorithm, but sometimes need to appear to the right. Another example occurs when lists of RTL items occur in a LTR sentence: the Bidi Algorithm will automatically assume that the order of items in the list should be <q dir="rtl">1, 2, 3</q>, but actually what's needed is <q dir="ltr">1, 2, 3</q>. In HTML, bidi isolation can be applied to a range of text by enclosing it in an element with a <code class="kw" translate="no">dir</code> attribute. In plain text there are Unicode formatting characters that can do the job. These mechanisms remove unwanted 'spillover effects'.</p>
+
+<p class="definition"><dfn>First-strong detection</dfn> is an algorithm that looks for the first strongly-directional character in a string, and then uses that to guess at the appropriate <a>base direction</a> for the string as a whole. Unicode code points are associated with properties relating to text direction: generally, letters in right-to-left scripts such as Arabic and Hebrew have a strong RTL direction, whereas Latin and Han characters have a strong LTR direction. Other characters, such as punctuation, only have a weak intrinsic directionality, and the actual directionality is determined according to the context in which they are found.</p>
+
+<p class="note">If you are unfamiliar with bidirectional or right-to-left text, there is a basic introduction <a href="https://www.w3.org/International/articles/inline-bidi-markup/uba-basics">here</a>. This will give you a basic grasp of how the <a>Unicode Bidirectional Algorithm</a> works and the interplay between it and the base direction, which will stand you in good stead for reading this document. Additional materials can be found in the Internationalization Working Group's <a href="https://www.w3.org/International/techniques/developing-specs.en?open=text_direction#text_direction">Techniques Index</a>.</p>
   
  <p class="definition"><dfn data-lt="natural language">Natural Language</dfn> The spoken, written, or signed communications used by human beings. [[LTLI]]</p>
```

Comment:
Terms such as this and several that follow, that do not begin a sentence, should have a full stop after, to make things clearer for people using voice browsers.

Reply:
Some definitions, such as Syntactic Content, point to LTLI or CHARMOD-NORM. I know that that's where these definitions came from, but if we're going to keep the links, i think we should point to The Glossary, instead.

Reply:
Okay.

Reply:
The first comment is a good one and I note that we don't actually define some of the terms. I have made of go of it just now.